martin/src/config/file/main.rs

```diff
@@ -66,11 +77,15 @@ pub struct Config {
     ///
     /// Can be overridden by [`tile_cache_size_mb`](Self::tile_cache_size_mb) or similar configuration options.
     pub cache_size_mb: Option<u64>,
+
     /// Maximum size of the tile cache in megabytes (0 to disable)
     ///
     /// Overrides [`cache_size_mb`](Self::cache_size_mb)
     pub tile_cache_size_mb: Option<u64>,
 
+    #[serde(default)]
+    pub on_invalid: OnInvalid,
```

Comment:
Lets do it this way.

The reason is that otherwise the files like `tests/exepected/auto_mini/save_config.yaml` contain this when serialized again.

```suggestion
    pub on_invalid: Option<OnInvalid>,
```